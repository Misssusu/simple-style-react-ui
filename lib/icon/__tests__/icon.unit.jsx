import renderer from "react-test-renderer";
import Icon from "../icon";
import React from "react";
import { mount } from "enzyme";


describe('icon',()=>{
    it('是一个 svg',()=>{
        const json = renderer.create(<Icon name="wechat"/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick',()=>{
        let n = 1;
        const fn = () =>{
            n = 2;
        }
        const component = mount(<Icon name="wechat" onClick={fn} />)
        component.find('svg').simulate('click')
        expect(n).toEqual(2)
    })
})