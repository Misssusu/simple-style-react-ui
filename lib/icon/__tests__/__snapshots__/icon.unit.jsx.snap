// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 是一个 svg 1`] = `
<svg
  className="s-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
